internal/channelz/funcs.go

```diff
@@ -143,21 +144,29 @@ func RegisterChannel(parent *Channel, target string) *Channel {
 // Returns a unique channelz identifier assigned to this subChannel.
 //
 // If channelz is not turned ON, the channelz database is not mutated.
-func RegisterSubChannel(pid int64, ref string) *SubChannel {
+func RegisterSubChannel(pid *Channel, ref string) (*SubChannel, error) {
+	if pid == nil {
+		return nil, errors.New("a SubChannel's parent id cannot be nil")
+	}
```

Comment:
Let's not return an error here since it complicates the code so much, and it should just be an invariant of the code like any other dereference.  Just panic and we'll make sure `RegisterSubChannel` is always called properly.

Actually, this change is pretty important to get done quickly, so I created #7101 and we will be doing a patch release with it ASAP.  Thank you for the fix!